tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -807,8 +819,15 @@ Status StorageManager::array_evolve_schema(
   return Status::Ok();
 }
 
-Status StorageManager::array_upgrade_version(
-    const ArrayDirectory& array_dir, const Config* config) {
+/*
+ * Note: `Config&` argument is not used, as it was no longer used after
+ * refactoring from `const Config* config`. A null check was deleted, after
+ * which a warning appear that the argument was not referenced. Encryption
+ * parameters are taken from `config_` and not from the former argument
+ * `config`, which is likely not what was intended.
+ */
```

Comment:
To me, this refactoring at whatever point occurring, on the surface at least, seems to have removed the (possibly not used) ability to over-ride the internal config_ with an externally provided config, for a given call of array_upgrade_version().  If it wasn't used, then likely irrelevant if not doc'd anywhere, but if used, what's happening in those situations now, what might that original refactoring have changed?

Reply:
This was done here: https://github.com/TileDB-Inc/TileDB/pull/3529
and the purpose was to ensure that the encryption key used to open the array matches the key used to upgrade. We do not support using different keys at different timestamps, so overriding the key here, when modifying it, would likely lead to an (at least partially) unreadable array.